framework/src/setroubleshoot/audit_data.py

```diff
@@ -332,19 +332,10 @@ def decode_fields(self):
                     decoded_value = audit_msg_decode(value)
                     self.fields[field] = decoded_value
 
-    def translate_path(self, path):
+    def translate_hex(self, path):
         try:
-            t = path.decode("hex")
-            if t[0].encode("hex") == "00":
-                tpath = "@"
-            else:
-                tpath = t[0]
-
-            for i in range(len(t))[1:]:
-                if t[i].encode("hex") != "00":
-                    tpath = tpath + t[i]
-                else:
-                    break
+            # produces str in python 3 and unicode string in python 2
+            tpath = bytearray.fromhex(path).decode('utf-8')
```

Comment:
You can directly do `return bytearray.fromhex(path).decode('utf-8')` here and remove the `return tpath` from the end of the function. It is shorter and also easier to read.